xla/service/gpu/buffer_allocations.cc

```diff
@@ -54,7 +54,23 @@ se::DeviceMemoryBase BufferAllocations::GetDeviceAddress(
     BufferAllocation::Index buffer_index) const {
   CHECK_GE(buffer_index, 0);
   CHECK_LT(buffer_index, buffers_.size());
-  return buffers_[buffer_index];
+  se::DeviceMemoryBase base = buffers_[buffer_index];
+  if (reinterpret_cast<uintptr_t>(base.opaque()) == kExternalAllocationMarker) {
+    if (!external_allocations_) {
+      LOG(ERROR) << "Does not have external allocations for buffer "
+                 << buffer_index;
+      return se::DeviceMemoryBase();
+    }
+    auto external_address =
```

Comment:
TF_ASSIGN_OR_RETURN

Reply:
TF_ASSIGN_OR_RETURN does not work here, right?  
It works if the parent functions needs a Status output, and the calling function also returns a Status